proof-systems/src/groth16/mod.rs

```diff
@@ -101,6 +149,36 @@ pub struct VerifyingKey<E: PairingEngine> {
     pub gamma_abc_g1:       Vec<E::G1Affine>,
 }
 
+impl<E: PairingEngine> VerifyingKey<E> {
+    fn check_gamma_abc_g1_points(gamma_abc_g1: &[E::G1Affine]) -> IoResult<()> {
```

Comment:
As for the Pedersen primitive: I do not see any use for this "sanity" check, as it does not guarantee security.